File: ocy/nodejs/basic-usage.ts
import { RivalzClientSdk } from "./RivalzClient";
import path from "path";
import fs from "fs/promises";
import * as dotenv from "dotenv";

dotenv.config();

async function main() {
  // Initialize the client
  console.log("🚀 Initializing Rivalz client...");
  const client = new RivalzClientSdk(process.env.RIVALZ_SECRET_TOKEN);

  try {
    // Example 1: Upload a file
    console.log("📁 Uploading file...");
    const uploadResult = await client.uploadFile(
      path.join(__dirname, "../documents/sample.pdf"),
      "sample.pdf"
    );
    console.log("✅ File uploaded successfully:", uploadResult);

    // Example 2: Upload a passport image
    console.log("\n📷 Uploading passport...");
    const passportResult = await client.uploadPassport(
      path.join(__dirname, "../documents/passport.jpg"),
      "passport.jpg"
    );
    console.log("✅ Passport uploaded successfully:", passportResult);

    // Example 3: Create a knowledge base
    console.log("\n📚 Creating knowledge base...");
    const knowledgeBase = await client.createRagKnowledgeBase(
      path.join(__dirname, "../documents/knowledge.pdf"),
      "My Knowledge Base"
    );
    console.log("✅ Knowledge base created:", knowledgeBase.id);

    // Example 4: Add document to knowledge base
    console.log("\n📄 Adding document to knowledge base...");
    const document = await client.addDocumentToKnowledgeBase(
      path.join(__dirname, "../documents/additional.pdf"),
      knowledgeBase.id
    );
    console.log("✅ Document added:", document.id);

    let response = await client.getKnowledgeBases();
    console.log("✅ Knowledge bases:", response);

    // Example 5: Create a chat session
    while (response[0].status !== "ready") {
      response = await client.getKnowledgeBases();
    }
    const knowledgeBaseId = response[0].id;
    console.log("\n💬 Starting chat session...");
    const chatResponse = await client.createChatSession(
      knowledgeBaseId,
      "What is the main topic of the document?"
    );
    console.log("✅ Chat response:", chatResponse);

    // Example 6: Get uploaded documents
    const documents = await client.getUploadedDocuments();
    console.log("✅ Documents:", documents);

    // Example 7: Get uploaded history
    const uploadedHistory = await client.getUploadedHistory(0, 20);
    console.log("✅ Uploaded history:", uploadedHistory);

    // Example 8: Download a file
    console.log("\n⬇️ Downloading file...");
    const downloadPath = await client.downloadFile(
      uploadedHistory.uploadHistories[0].uploadHash,
      path.join(__dirname, "../downloads")
    );
    console.log("✅ File downloaded to:", downloadPath);

    // Example 9: Delete all files
    await client.deleteFile(uploadedHistory.uploadHistories[0].uploadHash);
    console.log("✅ File deleted successfully");
  } catch (error) {
    console.error("❌ Error:", error instanceof Error ? error.message : error);
  }
}

// Run the example
if (require.main === module) {
  main().catch((error) => {
    console.error("❌ Fatal error:", error);
    process.exit(1);
  });
}

File: ocy/python/basic_usage.py
import os
import time
from pathlib import Path
from dotenv import load_dotenv
from rivalz_client_sdk import RivalzClientSdk

# Load environment variables
load_dotenv()

# Get current directory
CURRENT_DIR = Path(__file__).parent

def main():
    # Initialize the client
    print("🚀 Initializing Rivalz client...")
    client = RivalzClientSdk(os.getenv("RIVALZ_SECRET_TOKEN"))

    try:
        # Example 1: Upload a file
        print("📁 Uploading file...")
        upload_result = client.upload_file(
            CURRENT_DIR / "../documents/sample.pdf"
        )
        print("✅ File uploaded successfully:", upload_result)

        # Example 2: Upload a passport image
        print("\n📷 Uploading passport...")
        passport_result = client.upload_passport(
            CURRENT_DIR / "../documents/passport.jpg"
        )
        print("✅ Passport uploaded successfully:", passport_result)

        # Example 3: Create a knowledge base
        print("\n📚 Creating knowledge base...")
        knowledge_base = client.create_rag_knowledge_base(
            CURRENT_DIR / "../documents/knowledge.pdf",
            "My Knowledge Base"
        )
        print("✅ Knowledge base created:", knowledge_base["id"])

        # Example 4: Add document to knowledge base
        print("\n📄 Adding document to knowledge base...")
        document = client.add_document_to_knowledge_base(
            CURRENT_DIR / "../documents/additional.pdf",
            knowledge_base["id"]
        )
        print("✅ Document added:", document["id"])

        # Example 5: Get knowledge bases and wait for ready status
        response = client.get_knowledge_bases()
        print("✅ Knowledge bases:", response)

        while response[0]["status"] != "ready":
            response = client.get_knowledge_bases()
            time.sleep(1)  # Wait 1 second before checking again

        # Example 6: Create a chat session
        print("\n💬 Starting chat session...")
        chat_response = client.create_chat_session(
            response[0]["id"],
            "What is the main topic of the document?"
        )
        print("✅ Chat response:", chat_response)

        # Example 7: Get uploaded documents
        documents = client.get_uploaded_documents()
        print("✅ Documents:", documents)

        # Example 8: Get uploaded history
        uploaded_history = client.get_upload_history(1, 10)
        print("✅ Uploaded history:", uploaded_history)

        # Example 9: Download a file
        print("\n⬇️ Downloading file...")
        upload_hash = uploaded_history[1][0]['uploadHash']
        file_name = uploaded_history[1][0]['fileName']
        print("✅ Upload hash:", upload_hash)
        download_path = client.download(
            upload_hash,
            CURRENT_DIR / f"../downloads/{file_name}"
        )
        print("✅ File downloaded to:", download_path)

        # Example 10: Delete a file
        print("\n🗑️ Deleting file...")
        client.delete_file(upload_hash)
        print("✅ File deleted successfully")

    except Exception as error:
        print("❌ Error:", str(error))

if __name__ == "__main__":
    main() 
